haskellPackages: update stackage and hackage

maralorn opened the pull request:
### This Merge

This PR is the regular merge of the `haskell-updates` branch into `master`.

This branch is being continually built and tested by hydra at https://hydra.nixos.org/jobset/nixpkgs/haskell-updates. You may be able to find an up-to-date Hydra build report at [cdepillabout/nix-haskell-updates-status](https://github.com/cdepillabout/nix-haskell-updates-status).

We roughly aim to merge these `haskell-updates` PRs at least once every two weeks. See the @NixOS/haskell [team calendar](https://cloud.maralorn.de/apps/calendar/p/Mw5WLnzsP7fC4Zky) for who is currently in charge of this branch.

### haskellPackages Workflow Summary

Our workflow is currently described in [`pkgs/development/haskell-modules/HACKING.md`](https://github.com/NixOS/nixpkgs/blob/haskell-updates/pkgs/development/haskell-modules/HACKING.md).

The short version is this:
* We regularly update the Stackage and Hackage pins on `haskell-updates` (normally at the beginning of a merge window).
* The community fixes builds of Haskell packages on that branch.
* We aim at at least one merge of `haskell-updates` into `master` every two weeks.
* We only do the merge if the [mergeable](https://hydra.nixos.org/job/nixpkgs/haskell-updates/mergeable) job is succeeding on hydra.
* If a [`maintained`](https://hydra.nixos.org/job/nixpkgs/haskell-updates/maintained) package is still broken at the time of merge, we will only merge if the maintainer has been pinged 7 days in advance. (If you care about a Haskell package, become a maintainer!)

---

This is the follow-up to #149730. Come to [#haskell:nixos.org](https://matrix.to/#/#haskell:nixos.org) if you have any questions.

sternenseemann committed:
maintainers/scripts: test for haskell configuration overlays

The added nix expression allows maintainers to check for regressions in
the configuration overlays employed by haskellPackages and friends. The
reasoning behind this is that, if we add an override for something, it
should also build. To test this fact, we extract all attributes touched
by a configuration and obtain all relevant derivations corresponding to
it which can then be thrown into nix-build --keep-going.

I've been using this expression to verify configuration-ghc-9.2.x.nix
for a week or so which works quite well. The amount of stale overrides
in other configuration makes it a bit more painful for other use cases
at the moment.

maralorn committed:
all-cabal-hashes: 2021-12-08T21:36:39Z -> 2021-12-09T20:50:23Z

This commit has been generated by maintainers/scripts/haskell/update-hackage.sh

maralorn committed:
haskellPackages: regenerate package set based on current config

This commit has been generated by maintainers/scripts/haskell/regenerate-hackage-packages.sh

github-actions[bot] committed:
Merge master into haskell-updates

github-actions[bot] committed:
Merge master into haskell-updates

kocielnik committed:
haskellPackages.Monadoro: remove from broken pkgs

Package tests are fixed in the recent version (0.2.6.0).

github-actions[bot] committed:
Merge master into haskell-updates

cdepillabout committed:
Merge pull request #150245 from kocielnik/master

haskellPackages.Monadoro: remove from broken pkgs

cdepillabout committed:
haskellPackages: regenerate package set based on current config

This commit has been generated by maintainers/scripts/haskell/regenerate-hackage-packages.sh

github-actions[bot] committed:
Merge master into haskell-updates

sternenseemann committed:
haskellPackages: adjust overrides for attoparsec update

latest attoparsec: 0.14.2 -> 0.14.3

sternenseemann committed:
haskellPackages: adjust overrides for hspec update

latest hspec{,-core,-discover}: 2.9.3 -> 2.9.4

dalpd committed:
haskellPackages.pinch: Fix build by bumping network

github-actions[bot] committed:
Merge master into haskell-updates

cdepillabout committed:
Merge pull request #150613 from dalpd/fix-pinch

haskellPackages.pinch: Fix build by bumping `network`

Ericson2314 committed:
haskellPackages.{github,nix-thunk}: Fix

cdepillabout committed:
haskellPackages.hslua: disable tests when building with musl

github-actions[bot] committed:
Merge master into haskell-updates

sternenseemann committed:
haskell.packages.ghc901.ChasingBottoms: disable stale broken flag

sternenseemann committed:
haskell.compiler.ghc921.hlint: 3.3.4 -> unstable-2021-12-12

hlint introduced GHC 9.2.1 support on its development branch recently.

sternenseemann committed:
haskell.packages.ghc921.hoogle: unbreak

haskell.packages.ghc921.bsb-http-chunked: disable doctest suite

This package's doctest suite is invoked in a weird way apparently which
makes it fail to typecheck with GHC 9.2.1.

sternenseemann committed:
maintainers/scripts/test-configuration.nix: warn about broken pkgs

The warnings help when removing the broken flags when all packages turn
out to build and could also be helpful when investigating failures.

sternenseemann committed:
maintainers/scripts/test-configurations.nix: skip binary GHCs

While being able to test them is neat (on x86_64-linux they work very
well, actually), we usually don't want to do this, since the set is
only (recommended to be) used to bootstrap GHC. Consequently there is
almost no binary cache and testing them mostly leads to unenlightening
and seemingly endless compilation.

sternenseemann committed:
maintainers/scripts/test-configuration.nix: improve overlay call

configuration-nix.nix uses builtins.intersectAttrs to not any overrides
for packages not present in `super` (presumably for use outside of
nixpkgs?). To accomodate it, we pass an attribute set with every
attribute of haskellPackages, but set to `null` as `super`, and — while
we're at it — a fix point as `self`.

sternenseemann committed:
haskellPackages.http3: unmark as broken

Ericson2314 committed:
haskellPackages: regenerate package set based on current config

This commit has been generated by maintainers/scripts/haskell/regenerate-hackage-packages.sh

github-actions[bot] committed:
Merge master into haskell-updates

github-actions[bot] committed:
Merge master into haskell-updates

github-actions[bot] committed:
Merge master into haskell-updates

sternenseemann commented:
Note: Last update was 10 days ago, so we should finish this in the next few days to keep our promise of at least biweekly updates. Shouldn't be much work though, as this seems mostly done.

github-actions[bot] committed:
Merge master into haskell-updates

github-actions[bot] committed:
Merge master into haskell-updates

github-actions[bot] committed:
Merge master into haskell-updates

sternenseemann committed:
haskellPackages.Frames-streamly: drop now unnecessary override

Fix has been released by upstream.

sternenseemann committed:
haskell-ci-unstable: 0.13.20211011 -> 0.13.20211116-b61df11

Since version is not changed with every commit, let's append the git
revision to the version number.

sternenseemann committed:
Merge remote-tracking branch 'origin/master' into haskell-updates

sternenseemann committed:
haskellPackages: update list of transitively broken packages
